stateMutators.js

```diff
@@ -0,0 +1,174 @@
+const gameState = {
+    basketPos: 50,
+    basketValue: 0,
+    score: 0,
+    livesRemaining: 3,
+    gameLevel: 1,
+    gameMode: 'running',
+    fallingObjects: [],
+}
+
+const setBasketPos = (state, value) => {
+    return {
+        ...state,
+        basketPos: value,
+    }
+};
+
+const moveBasketLeft = (state, value) => {
+    return {
+        ...state,
+    basketPos: basketPos - value,
+    }
+};
+
+const moveBasketRight = (state, value) => {
+    return {
+        ...state,
+        basketPos: basketPos + value,
+    }
+};
+
+const setBasketValue = (state, value) => {
+    return {
+        ...state,
+        basketValue: value,
+    }
+};
+
+const resetBasketValue = (state) => {
+    return {
+        ...state,
+        basketValue: 0,
+    }
+};
+
+const setScore = (state, value) => {
+    return {
+        ...state,
+        score: value,
+    }
+};
+
+const resetScore = (state) => {
+    return {
+        ...state,
+        score: 0,
+    }
+};
+
+const loseLife = (state, value) => {
+    return {
+        ...state,
+        livesRemaining: value,
+    }
+};
+
+const resetLivesRemaining = (state) => {
+    return {
+        ...state,
+        livesRemaining: 3,
+    }
+};
+
+const setGameLevel = (state, value) => {
+    return {
+        ...state,
+        gameLevel: value,
+    }
+};
+
+const resetGameLevel = (state) => {
+    return {
+        ...state,
+        gameLevel: 0,
+    }
+};
+
+const setGameMode = (state, value) => {
+    return {
+        ...state,
+        gameMode: value,
+    }
+};
+
+const addFallingObject = (state, fallingObj) => {
+    return {
+        ...state,
+        fallingObjects: [
+            ...state.fallingObjects,
+            fallingObj,
+        ]
+    }
+};
+
+const removeFallingObject = (state, fallingObjId) => {
+    return {
+        ...state,
+        fallingObjects: [
+            ...state.fallingObjects.slice[0, fallingObjId],
+            ...state.fallingObjects.slice[fallingObjId + 1],
+        ]
+    }
+};
+
+
+// Object constructor for falling object
+function FallingObject(posX, posY, numerator, denominator) {
+    this.posX = posX;
+    this.posY = posY;
+    this.numerator = numerator;
+    this.denominator = denominator;
+    this.value = numerator / denominator;
+    this.id = id;
+}
+
+const setFallingObjPosY = (fallingObj, value) => {
```

Comment:
These ones should all return `fallingObj`. So they follow that convention of old state in, new state out. Even though this is not the full state, it still operates on a section of state and can be used easily in the same way. substate mutators

Reply:
updated